src/state/psarstake/hooks.ts

```diff
@@ -469,129 +515,139 @@ export function useDerivativeSarClaim(position: Position | null) {
-        .div(balance.isZero() ? 1 : balance);
-
-      if (!valueVariables || !rewardRate || !pendingRewards || !uri) {
-        return {} as Position;
-      }
+      const nftsIndexes = nfts?.map((index) => {
+        return [index.toHexString()];
+      });
+
+      // get the staked amount for each position via multicall
+      const positionsFragment = sarStakingContract.interface.getFunction('positions');
+      const positionsAmountCalls = nftsIndexes?.map((index) => {
+        return [sarStakingContract.address, sarStakingContract.interface.encodeFunctionData(positionsFragment, index)];
+      });
+
+      // get the reward rate for each position via multicall
+      const rewardRateFragment = sarStakingContract.interface.getFunction('positionRewardRate');
+      const rewardRateCalls = nftsIndexes?.map((index) => {
+        return [sarStakingContract.address, sarStakingContract.interface.encodeFunctionData(rewardRateFragment, index)];
+      });
 
-      return {
-        id: BigNumber.from(id),
-        balance: valueVariables?.balance,
-        sumOfEntryTimes: valueVariables?.sumOfEntryTimes,
-        apr: apr,
-        rewardRate: rewardRate,
-        uri: uri,
-        pendingRewards: pendingRewards,
-      } as Position;
-    });
-    // remove the empty positions
-    return { positions: positions.filter((position) => !!position), isLoading: false };
-  }, [account, balanceState, positionsAmountState, positionsRewardRateState, nftsState, nftsURIsState]);
+      // get peding rewards for each position via multicall
+      const pendingRewardsFragment = sarStakingContract.interface.getFunction('positionPendingRewards');
+      const pendingRewardsCalls = nftsIndexes?.map((index) => {
+        return [
+          sarStakingContract.address,
+          sarStakingContract.interface.encodeFunctionData(pendingRewardsFragment, index),
+        ];
+      });
+
+      // get all NFTs URIs from the positions via multicall
+      const nftUrisFragment = sarStakingContract.interface.getFunction('tokenURI');
+      const nftUrisCalls = nftsIndexes?.map((index) => {
+        return [sarStakingContract.address, sarStakingContract.interface.encodeFunctionData(nftUrisFragment, index)];
+      });
+
+      const results = await Promise.all([
+        multiCallContract.aggregate(positionsAmountCalls),
+        multiCallContract.aggregate(rewardRateCalls),
+        multiCallContract.aggregate(pendingRewardsCalls),
+        multiCallContract.aggregate(nftUrisCalls),
+      ]);
+
+      // decode the results
+      const positionsAmount = decodeMulticallResult(sarStakingContract, positionsFragment, results[0][1]);
+      const positionsRewardRate = decodeMulticallResult(sarStakingContract, rewardRateFragment, results[1][1]);
+      const positionsPendingRewards = decodeMulticallResult(sarStakingContract, pendingRewardsFragment, results[2][1]);
+      const nftsUris = decodeMulticallResult(sarStakingContract, nftUrisFragment, results[3][1]);
```

Comment:
`results[0][1]` is unsafe operation there are changes that code will crash, make sure you use null safe operation like `results?.[0]?.[1]`